pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -571,6 +574,16 @@ func (s *DiscoveryServer) pushConnection(con *XdsConnection, pushEv *XdsEvent) e
 		} else {
 			adsLog.Debugf("Skipping push to %v, no updates required", con.ConID)
 		}
+
+		if s.Env.StatusReporter != nil {
```

Comment:
Why would you report an event if no push happens ? You sure it's in the right place ?

Reply:
In this case, the proxy will never receive this particular version of the config, as the delta is not of interest to it.  From a distribution status perspective, this change should not count this envoy as 'not complete', so we register the event.

Reply:
And a comment would be perfect for future readers.

Reply:
Comment expanded.